crates/turborepo-paths/src/lib.rs

```diff
@@ -2,33 +2,43 @@
-/// representation. For reasons why, see [the `camino` documentation](https://github.com/camino-rs/camino/).
-///
-/// As in `std::path`, there are `Path` and `PathBuf` variants of each path
-/// type, that indicate whether the path is borrowed or owned.
-///
-/// When initializing a path type, it is highly recommended that you use a
-/// method that validates the path. This will ensure that the path is in the
-/// correct format. For the -Buf variants, the `new` method will validate that
-/// the path is either absolute or relative, and then convert it to either
-/// system or unix. For the non-Buf variants, the `new` method will *only*
-/// validate and not convert (this is because conversion requires allocation).
-///
-/// The only case where initializing a path type without validation is
-/// recommended is inside turbopath itself. But that unchecked initialization
-/// should be considered unsafe
+//! Turborepo's path handling library.
+//! Defines distinct path types for the different uses of paths in Turborepo's
+//! codebase.
+//!
+//! - `AbsoluteSystemPath(Buf)`: a path that is absolute and uses the system's
+//!   path separator. Used for interacting with the filesystem
+//! - `RelativeUnixPath(Buf)`: a path that is relative and uses the unix path
+//!   separator, i.e. `/`. Used when saving to a cache as a platform-independent
+//!   path.
+//! - `AnchoredSystemPath(Buf)`: a path that is relative to a specific directory
+//!   and uses the system's path separator. Used for handling files relative to
+//!   the repository root.
+//!
+//! NOTE: All paths contain UTF-8 strings. We use `camino` as the underlying
+//! representation for system paths. For reasons why, see [the `camino` documentation](https://github.com/camino-rs/camino/).
+//!
+//! As in `std::path`, there are `Path` and `PathBuf` variants of each path
+//! type, that indicate whether the path is borrowed or owned.
+//!
+//! # Validation
+//!
+//! When initializing a path type, it is highly recommended that you use a
+//! method that validates the path. This will ensure that the path is in the
+//! correct format. It's important to note that the validation will only
+//! check for absolute or relative and valid Unicode. It will not check
+//! if the path is system or unix, since that is not always decidable.
+//! For example, is `foo\bar` a Windows system path or a Unix path?
+//! It has a backslash, which is simultaneously a valid character in a
+//! Unix file name and the Windows path delimiter.
+//!
+//! Therefore, it's very important to keep the context of the initialization
+//! in mind. The only case where we do more in depth validation is for cache
+//! restoration. See `AnchoredSystemPathBuf::from_system_path` for more details.
+//!
+//! The only case where initializing a path type without validation is
+//! recommended is inside turbopath itself. But that unchecked initialization
+//! should be considered unsafe
```

Comment:
```suggestion
//! recommended is inside turbopath itself, but that unchecked initialization
//! should be considered unsafe.
```

Reply:
For dramatic effect :P

Reply:
![tenor](https://github.com/vercel/turbo/assets/490968/7ed83d45-d385-4e0e-8a53-3a327b656ea9)